compiler/src/iree/compiler/Codegen/LLVMGPU/TransformExtensions/LLVMGPUExtensionsOps.td

```diff
@@ -682,4 +682,44 @@ def PackSharedMemoryAllocOp :  Op<Transform_Dialect, "iree.pack_shared_memory_al
   }];
 }
 
+def TestAMDGPUContractionDistribution :
+  Op<Transform_Dialect, "iree.test_amdgpu_contraction_distribution",
+    [DeclareOpInterfaceMethods<MemoryEffectsOpInterface>,
+     TransformEachOpTrait,
+     TransformOpInterface,
+     ReportTrackingListenerFailuresOpTrait]> {
+  let description = [{
+    Run AMDGPU Vector Contraction distribution on the target as the root.
+
+    The anchor points are set by using the attribute
+    "__vector_layout_test_anchor_operand_x" and
+    "__vector_layout_test_anchor_result_x", where "x" is the operand/result
+    number.
+
+    This op produces amdgpu MFMA ops.
+
+    #### Return modes
+
+    This transform does not consume the target handle and always return success.
+    }];
+
+    let arguments = (ins TransformHandleTypeInterface:$target,
+                         I64Attr:$m,
+                         I64Attr:$n,
+                         I64Attr:$k);
```

Comment:
I feel like the layout should contain enough information to find m, n, k.

Reply:
For our use case, yes, but in general not necessarily.